### DISCUSSION TITLE
Usage question for interpolate_na

### DISCUSSION THREAD

**Discussion by alessioarena** [pydata/xarray#4639] [Q&A] [answered]
Hi All!

Just wondering what is the difference between the argument `max_gap` and `limit` in `interpolate_na`?

From the documentation:
max_gap - Maximum size of gap, a continuous sequence of NaNs, that will be filled
limit - Maximum number of consecutive NaNs to fill

http://xarray.pydata.org/en/stable/generated/xarray.DataArray.interpolate_na.html

**Comment by keewis** [comment, marked as answer]
consider this:
```python
In [2]: da = xr.DataArray(
   ...:     [0, np.nan, 1, np.nan, np.nan, 2, np.nan, np.nan, np.nan, 3],
   ...:     dims="x",
   ...:     coords={"x": np.arange(-10, 10, 2)},
   ...: )
   ...: da
Out[2]: 
<xarray.DataArray (x: 10)>
array([ 0., nan,  1., nan, nan,  2., nan, nan, nan,  3.])
Coordinates:
  * x        (x) int64 -10 -8 -6 -4 -2 0 2 4 6 8

In [3]: da.interpolate_na(dim="x", method="nearest", limit=2)
Out[3]: 
<xarray.DataArray (x: 10)>
array([ 0.,  0.,  1.,  1.,  2.,  2.,  2.,  2., nan,  3.])
Coordinates:
  * x        (x) int64 -10 -8 -6 -4 -2 0 2 4 6 8

In [4]: da.interpolate_na(dim="x", method="nearest", max_gap=7)
Out[4]: 
<xarray.DataArray (x: 10)>
array([ 0.,  0.,  1.,  1.,  2.,  2., nan, nan, nan,  3.])
Coordinates:
  * x        (x) int64 -10 -8 -6 -4 -2 0 2 4 6 8
```
which means that:
- `limit` constrains the number of consecutive missing values filled per gap (after which it stops)
- `max_gap` is the maximum size of the gap (in coordinate space) that should be filled. Any gap bigger than that will not be filled at all.

I agree that it's pretty difficult to figure that out reading just the parameter description. Maybe examples demonstrating that would make it easier to understand? If you're up for it, we would definitely welcome a PR adding those.

**Reply by dcherian** [reply]
:+1: for adding examples.

`limit` is copied from pandas. I added `max_gap` because I never found `limit`'s behaviour to be useful (pandas still has outstanding issues asking for `max_gap`).
